packages/next/src/server/app-render/encryption-utils.ts

```diff
@@ -86,43 +86,40 @@ export function setReferenceManifestsSingleton({
   }
 }
 
+export function unsetReferenceManifestsSingleton() {
+  // @ts-ignore
+  delete globalThis[SERVER_ACTION_MANIFESTS_SINGLETON]
+}
+
 export function getServerModuleMap() {
   const serverActionsManifestSingleton = (globalThis as any)[
     SERVER_ACTION_MANIFESTS_SINGLETON
-  ] as {
-    serverModuleMap: {
-      [id: string]: {
-        id: string
-        chunks: string[]
-        name: string
+  ] as
+    | {
+        serverModuleMap: {
+          [id: string]: {
+            id: string
+            chunks: string[]
+            name: string
+          }
+        }
       }
-    }
-  }
-
-  if (!serverActionsManifestSingleton) {
-    throw new Error(
-      'Missing manifest for Server Actions. This is a bug in Next.js'
-    )
-  }
+    | undefined
```

Comment:
Why would this be undefined? We always need a server module map.

Reply:
This type change is just reflecting how we currently handle it in `packages/next/src/server/load-components.ts`, where we don't set `globalThis[SERVER_ACTION_MANIFESTS_SINGLETON]` at all for route handlers. But I understand that we want to change this now. 👍